extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +81,21 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', [function() {
+  return {
+    Equals: function(answer, inputs) {
+      return answer.toLowerCase() == inputs.x.toLowerCase();
+    },
+    CaseSensitiveEquals: function(answer, inputs) {
+      return answer == inputs.x;
+    },
+    StartsWith: function(answer, inputs) {
+      return answer.indexOf(inputs.x, 0) == 0;
+    },
+    Contains: function(answer, inputs) {
+      return answer.toLowerCase().indexOf(inputs.x.toLowerCase(), 0) != -1;
+    }
+
```

Comment:
I think you'll need to add FuzzyEquals too -- it's only FuzzyMatches that is implicit. The former checks for a misspelling error and is actually an explicit rule. (We do have plans to remove this in the future and auto-check, actually, but let's just migrate them as-is for now to preserve backwards compatibility.)